schema/query.go

```diff
@@ -123,7 +123,7 @@ func validateQuery(q map[string]interface{}, validator Validator, parentKey stri
 				return nil, errors.New("$regex can only get strings as value")
 			}
 			if regex != "" {
-				queries = append(queries, Regex{Field: parentKey, Value: regex})
+				queries = append(queries, Regex{Field: parentKey, Value: regexp.MustCompile(regex)})
```

Comment:
Please use `Compile` instead and return the error if any. `MustCompile` would make the program to panic without much explanation.